LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs

```diff
@@ -71,15 +71,14 @@ internal static class DownlinkMessageBuilder
 
             if (receiveWindow == Constants.RECEIVE_WINDOW_2)
             {
-                tmst = rxpk.Tmst + timeWatcher.GetReceiveWindow2Delay(loRaDevice) * 1000000;
-
+                tmst = CalculateTime(rxpk.Tmst, timeWatcher.GetReceiveWindow2Delay(loRaDevice), loRaDevice.ReportedRXDelay);
```

Comment:
nit

tmst will always be rxpk.Tmst + delay.

The delay is computed based on the window we are plus device specific properties (as delaying the 1st window). For me, it is a bit more readable like this:

```c#
tmst = rxpk.Tmst + timeWatcher.GetWindowDelay(loRaDevice, receiveWindow);
```

You can now unit test all combinations of possible delays in the LoRaOperationTimeWatcher tests.